### PR TITLE
test: K8s test to expose an external IP address

### LINKED ISSUE #1200: K8s test to expose an external IP address
Add a test that exposes an external IP address to access an application

### PR DESCRIPTION
**Pull request by GabyCT**
This test exposes an external IP address to access a stateless aplication.

Fixes #1200

Signed-off-by: Gabriela Cervantes <gabriela.cervantes.tellez@intel.com>

### CODE REVIEW

File: integration/kubernetes/k8s-expose-ip.bats [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+	# Expose deployment
+	kubectl expose deployment/${deployment} --type=LoadBalancer --name=${service}
+
+	# Get service
+	kubectl get services ${service}
+	kubectl describe services ${service}
+
+	# Verify pods IP Addresses
+	IP_ADDRESSES=$(kubectl get pods --output=wide | awk '{print $6}' | tail -n +2)
```
Comment (grahamwhaley) [inline integration/kubernetes/k8s-expose-ip.bats]: Maybe we can do something more specific here using the `get pods -o ` JSON formatting, something like (but not, this is just a hacked up notion....):
```
kubectl get pods -o jsonpath='{.items[*].status.addresses[?(@.type=="ExternalIP")].address}'
```

**Comment by grahamwhaley** [comment]
I thought the CIs were mostly OK, but the 16.04 fail looks related:
```
23:29:38 not ok 1 Expose IP Address
23:29:38 # (in test file k8s-expose-ip.bats, line 29)
23:29:38 #   `kubectl wait --for=condition=Available deployment/${deployment}' failed
23:29:38 # INFO: k8s configured to use runtimeclass
23:29:38 # Image is up to date for gcr.io/google-samples/node-hello@sha256:d238d0ab54efb76ec0f7b1da666cefa9b40be59ef34346a761b8adc2dd45459b
23:29:38 # deployment.apps/hello-world created
23:29:38 # error: timed out waiting for the condition
23:29:38 # Error from server (NotFound): services "my-service" not found
23:29:38 # deployment.extensions "hello-world" deleted
23:29:38 Makefile:73: recipe for target 'kubernetes' failed
23:29:38 make: *** [kubernetes] Error 1
23:29:38 Build step 'Execute shell' marked build as failure
```

maybe some sort of timing error due to network slowness and container image update? I wonder if we can set the timeout on that 'wait' ?

File: integration/kubernetes/k8s-expose-ip.bats (hunk 1 above)
Comment (GabyCT) [reply integration/kubernetes/k8s-expose-ip.bats]: @grahamwhaley , thanks for the suggestion, I changed it to
```
kubectl get pods -o jsonpath='{.items[*].status.podIP}'
```

**Comment by jodh-intel** [comment]
Ping @kata-containers/tests - review needed over here :)

**Comment by grahamwhaley** [comment]
@jodh-intel - I think we have the acks, but need the CIs. The WIP check will need the label dropping. I'll nudge the 18.04 to see what happens....

File: integration/kubernetes/k8s-expose-ip.bats [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+	# Expose deployment
+	kubectl expose deployment/${deployment} --type=LoadBalancer --name=${service}
+
+	# Get service
+	kubectl get services ${service}
+	kubectl describe services ${service}
+
+	# Check pods are running
+	cmd="kubectl get pods -o jsonpath='{.items[*].status.phase}' | grep Running"
```
Comment (amshinde) [inline integration/kubernetes/k8s-expose-ip.bats]: @GabyCT You can instead use a `kubectl get pod "podname"` here to get the specific pod, instead on relying that there should be no other pods running. You can also query by label as well.

File: integration/kubernetes/k8s-expose-ip.bats [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+	waitForProcess "$wait_time" "$sleep_time" "$cmd"
+
+	# Verify pods IP Addresses
+	IP_ADDRESSES=$(kubectl get pods -o jsonpath='{.items[*].status.podIP}')
```
Comment (amshinde) [inline integration/kubernetes/k8s-expose-ip.bats]: Same suggestion as above.

**Comment by grahamwhaley** [comment]
afaict, this PR might be near ready to merge....
@GabyCT - can you check you have applied all/any feedback?
@amshinde can you review/ack for us?
@GabyCT - if/when you are ready, drop the WIP/DNM - as afaict that is the only thing stopping merge right now.

thx.

**Comment by amshinde** [comment]
@grahamwhaley @GabyCT I dont see all comments addressed yet.

File: integration/kubernetes/k8s-expose-ip.bats (hunk 2 above)
Comment (GabyCT) [reply integration/kubernetes/k8s-expose-ip.bats]: @amshinde , as this is a deployment, if you use 
```
$ kubectl get pod "hello-world"
Error from server (NotFound): pods "hello-world" not found
```
The pods have the name `hello-world` with a random termination
```
$ kubectl get pods
NAME                           READY   STATUS    RESTARTS   AGE
hello-world-648779775d-h2nh9   1/1     Running   0          20m
hello-world-648779775d-rwnkl   1/1     Running   0          20m
```
That is the main reason, I am getting the names of the pods like the above

File: integration/kubernetes/k8s-expose-ip.bats (hunk 3 above)
Comment (GabyCT) [reply integration/kubernetes/k8s-expose-ip.bats]: The same like above

**Comment by GabyCT** [comment]
@amshinde please see my coments thanks :)

**Comment by chavafg** [comment]
@GabyCT can you rebase and fix conflicts with master? Also if this is ready, please remove the DNM flag.

**Comment by grahamwhaley** [comment]
OK, I'm going to go a bit on trust here....
- @GabyCT has dropped the DNM
- The button is green
- I can't see the previous comments (thanks github!) , so, @amshinde - maybe a quick look over to ensure I didn't merge something that still had pending feedback to be applied

**Pull request merged**